src/frontends/pytorch/src/op/native_multi_head_attention.cpp

```diff
@@ -0,0 +1,195 @@
+    const auto value_transposed =
+        context.mark_node(std::make_shared<opset10::Transpose>(value_reshaped, qkv_transpose_dims));
+
+    const auto scale_one = context.mark_node(std::make_shared<opset10::ConvertLike>(one, query_transposed));
+    const auto scale_dim = context.mark_node(std::make_shared<opset10::ConvertLike>(embed_div_heads, query_transposed));
+    const auto scale_dim_sqrt = context.mark_node(std::make_shared<opset10::Sqrt>(scale_dim));
+    const auto scale = context.mark_node(std::make_shared<opset10::Divide>(scale_one, scale_dim_sqrt));
+
+    const auto transpose_dims =
+        context.mark_node(std::make_shared<opset10::Concat>(OutputVector{zero_1d, one_1d, three_1d, two_1d}, 0));
+    const auto key_transpose = context.mark_node(std::make_shared<opset10::Transpose>(key_transposed, transpose_dims));
+    const auto query_key_transpose_dot_product =
+        context.mark_node(std::make_shared<opset10::MatMul>(query_transposed, key_transpose));
+
+    auto scaled_dot_product =
+        context.mark_node(std::make_shared<opset10::Multiply>(query_key_transpose_dot_product, scale));
+
+    // Mask handling
+    if (!context.input_is_none(12)) {
+        auto atten_mask = context.get_input(9);
+        if (atten_mask.get_element_type() == element::boolean) {
+            const auto minus_inf_conv =
+                context.mark_node(std::make_shared<opset10::ConvertLike>(minus_inf, scaled_dot_product));
+            const auto mask_inverse = context.mark_node(std::make_shared<opset10::LogicalNot>(atten_mask));
+            atten_mask = context.mark_node(std::make_shared<opset10::ConvertLike>(atten_mask, scaled_dot_product));
+            atten_mask = context.mark_node(std::make_shared<opset10::Select>(mask_inverse, atten_mask, minus_inf_conv));
+        } else {
+            OPENVINO_THROW("NativeMultiHeadAttention does not (fully) support float masks");
+            atten_mask = context.mark_node(std::make_shared<opset10::ConvertLike>(atten_mask, scaled_dot_product));
+        }
+
+        if (context.const_input<int64_t>(12) == 1) {
+            const auto target_mask_reshape = context.mark_node(
+                std::make_shared<opset10::Concat>(OutputVector{batch_size, one_1d, one_1d, seq_size}, 0));
+            atten_mask = context.mark_node(std::make_shared<opset10::Reshape>(atten_mask, target_mask_reshape, false));
+        }
+
+        const auto target_mask_shape = context.mark_node(
+            std::make_shared<opset10::Concat>(OutputVector{batch_size, heads_1d, seq_size, seq_size}, 0));
+        atten_mask = context.mark_node(std::make_shared<opset10::Broadcast>(atten_mask, target_mask_shape));
+        scaled_dot_product = context.mark_node(std::make_shared<opset10::Add>(scaled_dot_product, atten_mask));
+    }
+
+    const auto scaled_dot_product_softmax =
+        context.mark_node(std::make_shared<opset10::Softmax>(scaled_dot_product, -1));
+    const auto scaled_dot_product_attention =
```

Comment:
It would be difficult / suboptimal, even though the core scaled dot product attention equation is the same, masks are applied differently and scale is different